ocaml/xapi/extauth_plugin_ADwinbind.ml

```diff
@@ -160,6 +168,20 @@ let clean_machine_account ~service_name = function
       debug
         "username or password not provided, skip cleaning the machine account"
 
+(* Clean local resources like machine password *)
+let clean_local_resources () : unit =
+  let folder = "/var/lib/samba/private" in
+  let secrets_tdb = Printf.sprintf "%s/secrets.tdb" folder in
+  try
+    (* Erase secrets database before clean the files *)
+    Helpers.call_script tdb_tool [secrets_tdb; "erase"] |> ignore ;
+    (* Clean local resource files *)
+    Xapi_fs.rm_tree ~rm_top:false folder
+  with e ->
+    let msg = "Failed to clean local samba resources" in
```

Comment:
The exception may raised by either `delete local folder`, or `erase samba database`, 
The error msg already has the details here, so let's just skip the folder name here. (as it may actually a erase database error)